src/Xamarin.Android.Build.Tasks/Tasks/GenerateResourceDesignerIntermediateClass.cs

```diff
@@ -0,0 +1,53 @@
+using System;
+using System.IO;
+using System.CodeDom.Compiler;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using Microsoft.Android.Build.Tasks;
+using MonoDroid.Tuner;
+
+namespace Xamarin.Android.Tasks
+{
+	public class GenerateResourceDesignerIntermediateClass : AndroidTask
+	{
+		public override string TaskPrefix => "GRDIC";
+
+		private const string ResourceDesigner = $"{FixLegacyResourceDesignerStep.DesignerAssemblyNamespace}.Resource";
+		private const string ResourceDesignerConstants = $"{FixLegacyResourceDesignerStep.DesignerAssemblyNamespace}.ResourceConstant";
+
+		private const string CSharpTemplate = @"// This is an Auto Generated file DO NOT EDIT
+using System;
+
+namespace %NAMESPACE% {
+	public partial class Resource : %BASECLASS% {
+	}
+}
+";
+		private const string FSharpTemplate = @"// This is an Auto Generated file DO NOT EDIT
+namespace %NAMESPACE%
+
+type Resource = %BASECLASS%
+";
```

Comment:
Does this feature obsolete https://github.com/xamarin/Xamarin.Android.FSharp.ResourceProvider?

Should we put something in the docs above about F#?

Reply:
We should as it does make the provider obsolete